packages/e2e/test/stake-pool/StakePoolProvider.test.ts

```diff
@@ -0,0 +1,220 @@
+  const pagination = { limit: 20, startAt: 0 };
+
+  let provider: StakePoolProvider;
+  let pools: Cardano.StakePool[] = [];
+
+  beforeAll(() => {
+    const env = envalid.cleanEnv(process.env, { STAKE_POOL_PROVIDER_URL: envalid.url() });
+    const config = { baseUrl: env.STAKE_POOL_PROVIDER_URL, logger };
+
+    provider = stakePoolHttpProvider(config);
+  });
+
+  it('without filters, some pools is found', async () => {
+    const { pageResults, totalResultCount } = await provider.queryStakePools({ pagination });
+    expect(totalResultCount).toBeGreaterThan(0);
+    pools = pageResults;
+  });
+
+  describe('filters and options', () => {
+    beforeAll(() => {
+      if (pools.length < 2)
+        throw new Error(
+          `'some pools is found' test didn't run or found only ${pools.length} pools, but at least 2 pools are required to run this test`
+        );
+    });
+
+    describe('identifier filter', () => {
+      it('using id as value', async () => {
+        const { totalResultCount } = await provider.queryStakePools({
+          filters: { identifier: { values: [{ id: pools[0].id }] } },
+          pagination
+        });
+        expect(totalResultCount).toBe(1);
+      });
+
+      it('using name as value', async () => {
+        const pool = pools.find(({ metadata }) => Boolean(metadata?.name));
+
+        if (!pool)
+          throw new Error("'some pools is found' test found only pools without name: this test can't be performed");
+
+        const { totalResultCount } = await provider.queryStakePools({
+          filters: { identifier: { values: [{ name: pool.metadata!.name }] } },
+          pagination
+        });
+        expect(totalResultCount).toBe(1);
+      });
+
+      it('using ticker as value', async () => {
+        const pool = pools.find(({ metadata }) => Boolean(metadata?.ticker));
+
+        if (!pool)
+          throw new Error("'some pools is found' test found only pools without ticker: this test can't be performed");
+
+        const { totalResultCount } = await provider.queryStakePools({
+          filters: { identifier: { values: [{ ticker: pool.metadata!.ticker }] } },
+          pagination
+        });
+        expect(totalResultCount).toBe(1);
+      });
+
+      it('using id OR ticker as value (same pool)', async () => {
+        const { totalResultCount } = await provider.queryStakePools({
+          filters: { identifier: { values: [{ id: pools[0].id }, { ticker: pools[0].metadata?.ticker }] } },
+          pagination
+        });
+        expect(totalResultCount).toBe(1);
+      });
+
+      it('using id OR ticker as value (distinct pools)', async () => {
+        const { totalResultCount } = await provider.queryStakePools({
+          filters: { identifier: { values: [{ id: pools[0].id }, { ticker: pools[1].metadata?.ticker }] } },
+          pagination
+        });
+        expect(totalResultCount).toBe(2);
+      });
+
+      it('using id AND ticker as value (same pool)', async () => {
```

Comment:
Would be nice to find and query to 2 pools here:
1. Has ticker
2. Doesn't have ticker